src/streamlink/plugins/tf1.py

```diff
@@ -42,16 +44,30 @@ def get_stream_urls(self, channel):
     def _get_streams(self):
         m = self.match
         if m:
-            channel = m.group(1) or m.group(2)
+            channel = ""
+            channel_id = ""
+            if m.group("live"):
+                channel = m.group("live")
+                channel_id = "L_" + channel.upper()
+            elif m.group("lci"):
+                channel = "LCI"
+                channel_id = "L_LCI"
+            elif m.group("stream"):
+                channel = m.group("stream")
+                channel_id = "L_FAST_v2l-" + m.group("stream")
             log.debug("Found channel {0}".format(channel))
-            for sformat, url in self.get_stream_urls(channel):
+            for sformat, url in self.get_stream_urls(channel_id):
                 try:
```

Comment:
The `try: ... except PluginError: ...` block should also be removed. It's unnecessary.

DASH streams and HLS multivariant streams should also not be yielded in a loop and instead the resulting dict of the `HLSStream.parse_variant_playlist` and `DASHStream.parse_manifest` should be returned by `Plugin._get_streams` and the loop be interrupted, so potential unnecessary/duplicate streams don't get included in the streams result.